src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs

```diff
@@ -54,10 +56,37 @@ public bool IsBrokerInstalledAndInvokable()
             }
         }
 
+        ///Check if the network is available. If the network is unavailable due to power optimization,
+        ///a <see cref="MsalClientException"/>will be thrown with error code <see cref="MsalError.AndroidBrokerClientPowerOptimization"/> 
+        private void CheckPowerOptimizationStatus()
+        {
+            var powerManager = PowerManager.FromContext(Application.Context);
+
+            //Power optimization checking was added in API 23
+            if (Build.VERSION.SdkInt == BuildVersionCodes.M &&
+                powerManager.IsDeviceIdleMode &&
+                !powerManager.IsIgnoringBatteryOptimizations(Application.Context.PackageName))
+            {
+                _logger.Warning("Power optimization detected on the client application: " + Application.Context.PackageName + " and the device is in doze mode or the app is standby. \n" +
```

Comment:
Does it make sense to make these errors? to ensure they show up in logs?

Reply:
I would include the error code w/it, at the beginning: `broker_no_network_connection_power_optimization` so people can easily search the logs and know what to look for quickly.

---
In reply to: [536436883](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2257#discussion_r536436883) [](ancestors = 536436883)

Reply:
Yes. using errors